var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -52,10 +55,43 @@ class Mesa(AutotoolsPackage):
     depends_on('libxfixes')
 
     depends_on('glproto@1.4.14:', type='build')
-    depends_on('dri2proto@2.6:', type='build')
-    depends_on('dri3proto@1.0:', type='build')
+    depends_on('dri2proto@2.6:', when='+dri', type='build')
+    depends_on('dri3proto@1.0:', when='+dri', type='build')
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
 
+    # adamjstewart
     # TODO: Add package for systemd, provides libudev
     # Using the system package manager to install systemd didn't work for me
+    # justiceformikebrown
+    # Added:
+    # ~dri will disable dri on older Centos /RHEL6 servers
+    # which lack newer udev
+    # You must also yum install libsysfs-devel
+    # https://www.mesa3d.org/osmesa.html
+    # https://www.mesa3d.org/autoconf.html
+    def configure_args(self):
+        spec = self.spec
+        args = []
+        if '~dri' in spec:
+            args.extend([
+                '--disable-dri',
+                '--disable-egl',
+                '--enable-sysfs',
+                '--disable-driglx-direct',
+                '--with-gallium-drivers=swrast'
+            ])
+        if '+dri' in spec:
+            args.extend([
+                '--enable-dri',
+                '--enable-egl'
+            ])
+
+        if '+osmesa' in spec:
+            args.extend([
+                '--enable-osmesa',
+                '--disable-driglx-direct',
+                '--disable-dri',
```

Comment:
You cannot do both.  This commit isn't doing both, either.  It's variant dri by default with osmesa disabled.

Demonstrating the mutual exclusivity:

`spack -d install mesa +dri+osmesa %gcc@6.3.0`

`--enable-dri' '--enable-egl' '--enable-osmesa' '--disable-driglx-direct' '--disable-dri' '--with-gallium-drivers=llvmpipe'
`
```
checking for PTHREAD_PRIO_INHERIT... no
checking for PTHREADSTUBS... yes
checking for LIBDRM... yes
checking for SHA1Init... no
checking for CC_SHA1_Init... no
checking for wincrypt.h... no
checking for SHA1Init in -lmd... no
checking for LIBSHA1... no
checking for nettle_sha1_init in -lnettle... no
checking for gcry_md_open in -lgcrypt... yes
checking for SHA1_Init in -lcrypto... yes
checking for OPENSSL... yes
checking for SHA1 implementation... libgcrypt
checking for LIBUDEV... no
checking for XLIBGL... yes
checking for mincore... yes
configure: error: egl requires --enable-dri
```